torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1975,35 +1975,78 @@ int listMulIntRight(Stack& stack) {
   return 0;
 }
 
+// Stolen (with appropriate modifications) from cpython repo
+// Objects/sliceobject.c with comment:
+// this is harder to get right than you might think
+//
+// This adjusts indexes according to python list semantics and returns number
+// of elements in the resulting list.
+static int64_t PySlice_AdjustIndices(
+    int64_t length, int64_t* start, int64_t* stop, int64_t step) {
+  TORCH_CHECK(step != 0, "List slice should have non-zero step")
```

Comment:
I simply didn't quite understand what they meant by 
assert(step >= -PY_SSIZE_T_MAX);
and why do we need the check in our implementation.

I believe there is already int64 boundaries checks when passing numbers from python to cpp?

Reply:
The point of this test is to prevent `step == INT64_MIN`. Because in that case `-step = step` (there is one more negative value than positive value and this - will overflow back to INT64_MIN) and line 2015 below will not work as expected.
I think you want to add it back, stating that a step of `INT64_MIN` is not supported (and hopefully no one will ever see that error message! )

Reply:
Done and added tests.

As we've looked with Alban, the overall behavior is actually correct without this check, but let's be safe and compatible with cpython :)